InvoiceWorker/InvoiceFileGenerator.cs

```diff
@@ -0,0 +1,43 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using InvoiceWorker.Models;
+
+namespace InvoiceWorker
+{
+    public interface IInvoiceGenerator
+    {
+        Task AddRecord(InvoiceEvent record);
+    }
+    
+    public class InvoiceFileGenerator : IInvoiceGenerator
+    {
+        private string GetInvoiceFilePath (Guid invoiceId) => $@"C:\InvoiceData-{invoiceId}";
+
+        public async Task AddRecord(InvoiceEvent invoiceEvent)
+        {
+            var lines = new List<string>
+            {
+                $"Invoice Number: {invoiceEvent.Content.InvoiceNumber}",
+                $"Status: {invoiceEvent.Content.Status}",
+                $"Created Date: {invoiceEvent.Content.CreatedDateUtc}",
+                $"Due Date: {invoiceEvent.Content.DueDateUtc}",
+                Environment.NewLine
```

Comment:
Hm would be good to have an OS specific one and see if they change it but then I guess it wouldn't be compatible on some candidate's machines